apoc.generate.er fails with IndexOutOfBoundsException

## Expected Behavior (Mandatory)
apoc.generate.er goes through and generates the graph

## Actual Behavior (Mandatory)
* nodes are created, but the procedure fails before creating any relationship.
* Neo.ClientError.Procedure.ProcedureCallFailed : 
"Failed to invoke procedure `apoc.generate.er`: Caused by: java.lang.IndexOutOfBoundsException: Index -559948134 out of bounds for length 510000"

## How to Reproduce the Problem
```
call apoc.generate.er(510000,750000, 'Node','HAS')
```

### Steps (Mandatory)

  1. call cypher above
  

## Screenshots (where it's possibile)
<img width="889" alt="er_bug" src="https://user-images.githubusercontent.com/45878183/101344695-c610f680-387d-11eb-84b6-989264f0234f.png">

## Specifications (Mandatory)
### Versions

  - OS: macOS
  - Neo4j: 4.2.1 (running locally in desktop)
  - Neo4j-Apoc:4.2.0.0

--- a/core/src/main/java/apoc/generate/relationship/ErdosRenyiRelationshipGenerator.java
+++ b/core/src/main/java/apoc/generate/relationship/ErdosRenyiRelationshipGenerator.java
@@ -37,7 +37,7 @@ public class ErdosRenyiRelationshipGenerator extends BaseRelationshipGenerator<E
     @Override
     protected List<Pair<Integer, Integer>> doGenerateEdges() {
         long threshold = getConfiguration().getNumberOfEdges() * 4;
-        long potentialEdges = getConfiguration().getNumberOfNodes() * (getConfiguration().getNumberOfNodes() - 1);
+        long potentialEdges = Long.valueOf(getConfiguration().getNumberOfNodes()) * (Long.valueOf(getConfiguration().getNumberOfNodes()) - 1);
 
         if (threshold > potentialEdges) {
             return doGenerateEdgesWithOmitList(); // Make sure to avoid edges
@@ -87,7 +87,7 @@ public class ErdosRenyiRelationshipGenerator extends BaseRelationshipGenerator<E
     private List<Pair<Integer, Integer>> doGenerateEdgesWithOmitList() {
         final int numberOfNodes = getConfiguration().getNumberOfNodes();
         final int numberOfEdges = getConfiguration().getNumberOfEdges();
-        final long maxEdges = numberOfNodes * (numberOfNodes - 1) / 2;
+        final long maxEdges = Long.valueOf(numberOfNodes) * (Long.valueOf(numberOfNodes) - 1) / 2;
 
         final List<Pair<Integer, Integer>> edges = new LinkedList<>();
 

